backend/modules/publishing/service.py

```
@@ -542,10 +565,144 @@ async def save_simulation_draft(
+                            continue
+                        # Generate tabular preview (headers + 5 rows)
+                        preview = file_info.get("preview", "")
+                        if not preview:
+                            preview = self._generate_data_preview(filename, file_bytes)
+                        updated_data_files.append({
+                            "filename": filename,
+                            "s3_key": s3_key,
+                            "preview": preview,
+                        })
+                        logger.info(f"[SAVE] Uploaded data file '{filename}' to S3: {s3_key}")
+                    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)